vcd/datasource_vcd_nsxt_edgegateway.go

```diff
@@ -105,27 +116,115 @@ func datasourceVcdNsxtEdgeGatewayRead(ctx context.Context, d *schema.ResourceDat
 	log.Printf("[TRACE] NSX-T edge gateway datasource read initiated")
 
 	vcdClient := meta.(*VCDClient)
-
-	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	org, err := vcdClient.GetOrgFromResource(d)
 	if err != nil {
 		return diag.FromErr(fmt.Errorf("error retrieving Org: %s", err))
 	}
 
-	if vdc.IsNsxv() {
-		return diag.Errorf("please use 'vcd_edgegateway' for NSX-V backed VDC")
-	}
+	// Validate if VDC or VDC Group is NSX-T backed
+	inheritedVdcField := vcdClient.Vdc
+	vdcField := d.Get("vdc").(string)
+	ownerIdField := d.Get("owner_id").(string)
 
-	edge, err := vdc.GetNsxtEdgeGatewayByName(d.Get("name").(string))
+	err = valdateIfVdcOrVdcGroupIsNsxt(org, inheritedVdcField, vdcField, ownerIdField)
 	if err != nil {
-		return diag.FromErr(fmt.Errorf("could not retrieve NSX-T edge gateway: %s", err))
+		return diag.FromErr(err)
+	}
+
+	var edge *govcd.NsxtEdgeGateway
+	edgeGatewayName := d.Get("name").(string)
+	ownerId := d.Get("owner_id").(string)
+	switch {
+	case ownerId != "":
```

Comment:
Just curiosity, why not use a `if-else`?

Reply:
I usually put such cases to have a `default`with error - to be sure that all cases are matched. This was left out so a good spot.

Another think led me to change the clause at all and narrow down the search by name to `vdc`(previously was scoped to `org`). An `Org` can have multiple Edge Gateways with the same name (in different VDCs).